src/plugins/plugin.legend.js

```diff
@@ -170,7 +170,9 @@ export class Legend extends Element {
     let row = -1;
     let top = -lineHeight;
     me.legendItems.forEach((legendItem, i) => {
-      const itemWidth = boxWidth + (fontSize / 2) + ctx.measureText(legendItem.text).width;
+      const pointStyle = legendItem?.pointStyle;
+      const pointWidth = pointStyle?.offsetWidth || pointStyle?.width || boxWidth;
```

Comment:
Would make sense to render this to the DOM in a `visibility: hidden` element, get offsetWidth, then remove from dom

Reply:
I think we should avoid DOM manipulation.